library/Zend/Log/Writer/Db.php

```diff
@@ -163,7 +163,12 @@ protected function mapEventIntoColumn(array $event, array $columnMap = null)
             if (is_array($value)) {
                 foreach ($value as $key => $subvalue) {
                     if (isset($columnMap[$name][$key])) {
-                        $data[$columnMap[$name][$key]] = $subvalue;
+                        if (is_scalar($subvalue)) {
+                            $data[$columnMap[$name][$key]] = $subvalue;
+                            continue;
+                        }
+
+                        $data[$columnMap[$name][$key]] = var_export($subvalue, true);
```

Comment:
When using `json_encode()` objects are stored as stdClass and not as the original class name and this causes loss of data. I looked at other possibilities too (see this comment  https://github.com/zendframework/zf2/issues/7244#issuecomment-76534596 and the one after that).

`var_export()` keeps the data readable. If the data needs to be processed further, the method [`Zend\Log\Formatter\Base::normalize()`](https://github.com/zendframework/zf2/blob/67f098af070b29d5042e89e936604df3193d2212/library/Zend/Log/Formatter/Base.php#L72) provides a nice way to do this, but I don't see how I can use that method (see my previous comment https://github.com/zendframework/zf2/pull/7282#issuecomment-76558776).